cheatsheets/Cryptographic_Storage_Cheat_Sheet.md

```diff
@@ -1,234 +1,175 @@
+  * This could also be caused by a someone who had access to the key leaving the organisation.
+- After a specified period of time has elapsed (known as the cryptoperiod).
+  * There are many factors that could affect what an appropriate cryptoperiod is, including the size of the key, the sensitivity of the data, and the threat model of the system. See section 5.3 of [NIST SP 800-57](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-57pt1r4.pdf) for further guidance.
+- After the key has been used to encrypt a specific amount of data.
+  * This would typically be `2^35` bytes (~34GB) for 64-bit keys (DES, 3DES, Blowfish, RC5, etc.) and `2^68` bytes (~295 exabytes) for 128 bit keys (AES, Twofish, Serpent, etc.).
+- If there is a significant change to the security provided by the algorithm (such as a new attack being announced).
 
-The method used to distribute keys must be secure to prevent the theft of keys in transit. The use of a protocol such as Diffie Hellman can help secure the distribution of keys, the use of secure transport such as TLS and SSHv2 can also secure the keys in transit. Older protocols like SSLv3 should not be used.
+Once one of these criteria have been met, a new key should be generated and used for encrypting any new data. There are two main approaches for how existing data that was encrypted with the old key(s) should be handled:
 
-###### 3.6.3 Secure cryptographic key storage
+- Decrypting it and re-encrypting it with the new key.
+- Marking each item with the ID of the key that was used to encrypt it, and storing multiple keys to allow the old data to be decrypted.
 
-The secure storage of encryption keys including KEK's has been touched on in our description of requirement 3.5 (see above).
+The first option should generally be preferred, as it greatly simplifies both the application code and key management processes; however, it may not always be feasible. Note that old keys should generally be stored for a certain period after they have been retired, in case old backups of copies of the data need to be decrypted.
 
-###### 3.6.4 Periodic cryptographic key changes
+It is important that the code and processes required to rotate a key are in place **before** they are required, so that keys can be quickly rotated in the event of a compromise. Additionally, processes should also be implemented to allow the encryption algorithm or library to be changed, in case a new vulnerability is found in the algorithm or implementation.
 
-The PCI DSS standard mandates that keys used for encryption must be rotated at least annually. The key rotation process must remove an old key from the encryption/decryption process and replace it with a new key. All new data entering the system must encrypted with the new key. While it is recommended that existing data be rekeyed with the new key, as per the Rekey data at least every one to three years rule above, it is not clear that the PCI DSS requires this.
+## Key Storage
```

Comment:
It would be good to briefly describe why HSM/KMS itp is better than just encrypted file on disk

Reply:
I've added some notes about the benefits of KMS/HSM solutions.